src/clj/puppetlabs/puppetserver/ringutils.clj

```diff
@@ -85,6 +109,28 @@
         false))
     true))
 
+(schema/defn ^:always-validate request->name :- (schema/maybe schema/Str)
+  "Return the name of the client by parsing the subject of the client cert from
+  the request.  Return nil if the client cert is absent or cannot be parsed."
+  [request :- RingRequest]
+  (let [{:keys [ssl-client-cert]} request]
+    (if ssl-client-cert (get-cert-subject ssl-client-cert))))
+
+(schema/defn ^:always-validate authorized? :- schema/Bool
+  "(SERVER-768) replacement for client-allowed-access?"
+  [rules :- rules/Rules
+   request :- RingRequest]
+  (if-let [name (request->name request)]
```

Comment:
Although if we go with the approach of allowing the `name` to be derived from a reverse lookup on `:remote-addr` as a fallback that we'd still also need something like the "auth" keyword from the old 'auth.conf' - https://docs.puppetlabs.com/puppet/latest/reference/config_file_auth.html#auth - in order to inform per rule as to whether or not an "unauthenticated" request should be allowed.

Where the fallback `:remote-addr` reverse lookup seems like it could open up a security hole for a client to trick the master into trusting it based on its ip address being that of a node that should legitimately be trusted, seems like it might be better to not carry forward with the fallback lookup.  Curious to get @ahpook's thoughts on that.

In any event, still a good discussion to have about whether we need to introduce some sort of a "authorized if actually authentic" distinction into the definition of the auth rules.  This might also inform how we approach compatibility for the use of `X-Client-*` headers for plaintext connections.